pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -80,9 +84,9 @@
   private final long _startTimeMillis = System.currentTimeMillis();
 
   private final String _segmentName;
-  private final Schema _schema;
+  private final Schema _originalSchema;
```

Comment:
Given that we dont have a member called `_newSchema`, why rename this? Can we keep it as `_schema`?

Reply:
Would like to avoid any potential confusion. If a schema update happens on the consuming segment, segmentMetadata and schema are still the original ones, without including the newly added columns

Reply:
this feels like leaking information about very specific behavior into the MutableSegmentImpl. Instead, in the addExtraColumns method, the Schema param can be renamed to "newSchema". Besides, you already have _newColumnsFieldSpecMap